packages/solecs/v2/tables/Vec2Table.sol

```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { console } from "forge-std/console.sol";
+import { IStore } from "../IStore.sol";
+import { StoreSwitch } from "../StoreSwitch.sol";
+import { StoreCore } from "../StoreCore.sol";
+import { SchemaType } from "../Types.sol";
+import { Bytes } from "../Bytes.sol";
+
+// -- User defined schema and id --
+
+bytes32 constant id = keccak256("mud.store.table.vec2");
+
+struct Schema {
+  uint32 x;
+  uint32 y;
+}
```

Comment:
what would a scalar look like in this model? would we always have to pass around a single-value struct and call `.value` on it (like we do on the client)? is there a more ergonomic approach?

Reply:
I think it's entirely possibly to make a special case for scalars, which would be entirely autogen's responsibility.
`Schema` is more of a convenience thing for users - StoreCore doesn't care about it.